### DISCUSSION TITLE
Confused about v40's Credential Provider

### DISCUSSION THREAD

**Discussion by gutley** [mozilla-mobile/firefox-ios#9570] [Q&A]
Just starting to play around with the credential provider and how it interacts with Firefox. One thing I don't understand is if I don't want to use Firefox for password autofill at all and just want to keep using Keychain/iCloud Keychain, then logging in to websites via Firefox never seems to save the password. If I manually add them in Settings app, then I can auto-fill them, but I never seem to be able to auto-save them?

Here's how I have my password settings configured in Settings App:
![IMG_03E4E9C821A8-1](https://user-images.githubusercontent.com/2040690/143903273-d261e8d3-fb1c-405e-bc56-48e8fd387016.jpeg)

Am I misunderstanding what this is all supposed to do?

**Comment by gutley** [comment]
BTW - I'm referring to Firefox above, but am really using Fennec for testing.

**Comment by SimonBasca** [comment]
@gutley logging in to Firefox should offer to save password even without enabling auto fill for Firefox in iOS Settings. This might be a bug. Could you please provide a few example links where Firefox doesn't offer to save passwords.

**Reply by gutley** [reply]
@SimonBasca One example where it doesn't ask nor save - instagram.com

**Reply by gutley** [reply]
Just to clarify, this is what happens after signing in on Instagram (and other site) using Safari. I have never seen this in Firefox (either in release version or during v40 branch.)
![IMG_4145](https://user-images.githubusercontent.com/2040690/145235507-e467fd9e-cb52-4346-95e8-8e63b2d16133.PNG)
:

**Reply by SimonBasca** [reply]
Could be related to this issue https://github.com/mozilla-mobile/firefox-ios/issues/8743

**Reply by gutley** [reply]
@SimonBasca - OK - I have now managed to find some sites where the FF equivalent of the above alert shows up and offers to save the password. But that leads to another issue that I am not sure is a FF bug or a limitation of how Credential Providers work.

Basically, when Firefox is not ticked in the Autofill Password Settings screen, saving a password in FF still just saves it to FF's own LoginsStore and not any of the other credential providers. As such, it becomes unavailable the next time you try to log in to the same site because the OS doesn't attempt to check with the FF credential provider as its not ticked, so it can't autofill. Going to FF's own Passwords screen will show the user/name password combo.
